File: app/src/main/java/com/noahkim/rolltime/data/Video.java
package com.noahkim.rolltime.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.database.Exclude;
import com.google.firebase.database.IgnoreExtraProperties;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Noah on 10/5/2017.
 */

@IgnoreExtraProperties
public class Video implements Parcelable {

    private String mTitle;
    private String mThumbnail;
    private String mVideoId;

    protected Video(Parcel in) {
        mTitle = in.readString();
        mThumbnail = in.readString();
        mVideoId = in.readString();
    }

    public Video() {
    }

    public Video(String title, String thumbnail,String videoId) {
        mTitle = title;
        mThumbnail = thumbnail;
        mVideoId = videoId;
    }

    @Exclude
    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put("title", mTitle);
        result.put("thumbnail", mThumbnail);
        result.put("videoId", mVideoId);

        return result;
    }

    public static final Creator<Video> CREATOR = new Creator<Video>() {
        @Override
        public Video createFromParcel(Parcel in) {
            return new Video(in);
        }

        @Override
        public Video[] newArray(int size) {
            return new Video[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(mTitle);
        parcel.writeString(mThumbnail);
        parcel.writeString(mVideoId);
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        mTitle = title;
    }

    public String getThumbnail() {
        return mThumbnail;
    }

    public void setThumbnail(String thumbnail) {
        mThumbnail = thumbnail;
    }

    public String getVideoId() {
        return mVideoId;
    }

    public void setVideoId(String videoId) {
        mVideoId = videoId;
    }
}

File: app/src/main/java/com/noahkim/rolltime/model/Video.kt
package com.noahkim.rolltime.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Noah on 10/5/2017.
 */

class Video protected constructor(`in`: Parcel) : Parcelable {

    var title: String? = null
    var thumbnail: String? = null
    var videoId: String? = null

    init {
        title = `in`.readString()
        thumbnail = `in`.readString()
        videoId = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(title)
        parcel.writeString(thumbnail)
        parcel.writeString(videoId)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<Video> = object : Parcelable.Creator<Video> {
            override fun createFromParcel(`in`: Parcel): Video {
                return Video(`in`)
            }

            override fun newArray(size: Int): Array<Video?> {
                return arrayOfNulls(size)
            }
        }
    }
}
